awscrt/io.py

```diff
@@ -96,28 +87,22 @@ def on_shutdown():
 
     @staticmethod
     def get_or_create_static_default():
-        global _static_event_loop_group
-        if '_static_event_loop_group' in globals():
-            if isinstance(_static_event_loop_group, EventLoopGroup):
-                return _static_event_loop_group
-
-        os_cpu_count = os.cpu_count()
-        if os_cpu_count is None:
-            os_cpu_count = 1  # could not find the processor count - default to 1
-        else:
-            # os.cpu_count returns all cores on the system (including virtual ones)
-            # so we divide by two to avoid using all cores on the entire system.
-            os_cpu_count = max(1, int(os_cpu_count / 2))
-
-        _static_event_loop_group = EventLoopGroup(os_cpu_count)
-        return _static_event_loop_group
+        EventLoopGroup._static_event_loop_group_lock.acquire()
+        if EventLoopGroup._static_event_loop_group is not None:
+            if isinstance(EventLoopGroup._static_event_loop_group, EventLoopGroup):
+                EventLoopGroup._static_event_loop_group_lock.release()
+                return EventLoopGroup._static_event_loop_group
+        EventLoopGroup._static_event_loop_group = EventLoopGroup()
+        EventLoopGroup._static_event_loop_group_lock.release()
+        return EventLoopGroup._static_event_loop_group
```

Comment:
as an aside: this still isn't thread safe, though it won't matter any more once you use the with statement

it's theoretically possible that someone calls `EventLoopGroup.release_static_default()` from another thread in between the lock being released, and `EventLoopGroup._static_event_loop_group` being returned. In which case `None` would end up being returned.

The thread safe way to do it when manually releasing a lock would be like:
```
tmp = EventLoopGroup._static_event_loop_group
EventLoopGroup._static_event_loop_group_lock.release()
return tmp
``` 

so that even if the value of `EventLoopGroup._static_event_loop_group` changes them *moment* you release the lock, you still return what you'd meant to return